mlir-sycl/lib/Conversion/SYCLToSPIRV/SYCLToSPIRV.cpp

```diff
@@ -294,9 +246,7 @@ class SingleDimGridOpPattern : public GridOpPattern<OpTy> {
 template <typename... OpTys>
 void addGridOpPatterns(RewritePatternSet &patterns,
                        TypeConverter &typeConverter, MLIRContext *context) {
-  (patterns.add<GridOpPatternIndex<OpTys>, GridOpPatternNoIndex<OpTys>>(
-       typeConverter, context),
-   ...);
+  (patterns.add<GridOpPatternNoIndex<OpTys>>(typeConverter, context), ...);
```

Comment:
As now we only have one pattern, should we just remove addGridOpPatterns? and do something like:
```
  patterns.add<GridOpPatternNoIndex<SYCLGlobalOffsetOp>,
               GridOpPatternNoIndex<SYCLNumWorkGroupsOp>,
               .......>(typeConverter, context);
```

Reply:
ue. Done that and renamed the pattern to make more sense.